r-package/grf/vignettes/rate.Rmd

```diff
@@ -0,0 +1,216 @@
+
+## RATE: The area under the TOC
+This picture suggests a quite natural way to asses the value of treating every according to $S(X_i)$: sum up the area under the curve. This is exactly how the **RATE** is defined, it is the *A*rea *U*nder the *TOC* ("AUTOC")
+
+$$\textrm{RATE} = \int_0^1 \textrm{TOC}(q) dq .$$
+If there is barely any heterogeneity in $\tau(X_i)$ this area will be vanishingly small and in the special case where $\tau(X_i)$ is constant, it's zero. Thinking one step ahead, if our *estimated* rule $S(X_i)$ do well in identifying the individuals with very different benefit of treatment, we would expect this metric to be *positive*. Conversely, if they do badly, or there barely are any benefit to stratifying treatment, we would expect it to be *negative* or close to *zero*, respectively.
+
+```{r toc-plot-autoc, echo=FALSE}
+n <- 1000
+p <- 1
+X <- matrix(rnorm(n * p), n, p)
+tau <- X[, 1]
+
+ATE <- mean(tau)
+sort.idx <- order(tau, decreasing = TRUE)
+TOC <- rep(NA, n)
+for (i in 1:n) {
+  TOC[i] <- mean(tau[sort.idx[1:i]]) - ATE
+}
+q <- seq(1/n, 1, by = 1/n)
+df <- data.frame(q, TOC)
+
+ggplot(df, aes(x = q, y = TOC)) +
+  geom_line() +
+  geom_hline(yintercept = 0, lty = 3) +
+  theme_classic() +
+  labs(y = "", title = "Targeting Operator Characteristic") +
+  geom_ribbon(aes(ymin = 0), ymax=TOC, alpha = 0.5, fill = "red") +
+  annotate(geom = 'text', x = 0.15, y = 0.6, color = 'black', hjust = -0.1,
+           label = "RATE")
+```
+
+There are different ways this area can be summed up, which gives rise to different RATE metrics. We refer to the integral above as the "AUTOC". This implies placing more weight on areas under the curve where the expected treatment benefit is largest. When non-zero treatment effects are concentrated among a small subset of the population this weighting has greater power for testing against a sharp null (RATE=0) as upweighting the individuals with a non-zero treatment effect increases the RATE.
+
+Another way to sum this area would be to weight each point on the curve by $q$: $\int_0^1 q \textrm{TOC}(q) dq$. We refer to this metric as the "QINI" (Radcliffe, 2007). This weighting implies placing as much weight on units with low treatment effects as on units with high treatment effects. When non-zero treatment effects are more diffuse across the entire population, this weighting tends to give greater power when testing against a null effect.
+
+Another thing to note is that the "high-vs-low" out-of-bag CATE quantile construction used in Athey and Wager (2019) can also be seen as a kind of RATE, where in place of Qini's identity weighting it would have point mass at q = a given quantile.
+
+## Estimating the RATE
```

Comment:
In general, I'm not sure what to think of "the RATE" vs "a RATE" (as RATEs are a class of metrics that require one to specify the weighing function, and a class of estimators that requires a specific weighting function _and_ CATE score).

Reply:
I only wanted to get Athey and Wager (2019) into the vignette since it is a widely popular way people "check" for HTEs using causal forest and it seems to me RATE can replace this exercise? 

I am not exactly sure what the exact formal equivalence is, but I do suggest we get a note on this into the vignette, otherwise users may end up asking things like: should we still do quantile sorts ala Athey/Wager/etc? Maybe you a have good formulation in mind?

Reply:
Also, as a bonus, if possible it would be great with a one-liner relating RATE to what the [test_calibration](https://grf-labs.github.io/grf/reference/test_calibration.html) "omnibus" test tells you? (I still don't get exactly what that word means here...). You and Stefan knows best, just asking if you have anything in mind! (trying to pre-empt future questions users are going to ask about rate)